File: 241008/제거한 숫자의 최솟값/minimum-of-numbers-removed.swift
import Foundation

let n = Int(readLine()!)!
let nums = readLine()!.split(separator: " ").map { Int64($0)! }.sorted()
var m: Int64 = 0

for i in 0..<n / 2 {
    m = max(m, nums[i] + nums[n - i - 1])
}

if n % 2 == 1 {
    m = max(m, nums[n / 2])
}

print(m)

File: 241211/제거한 숫자의 최솟값/minimum-of-numbers-removed.py
import sys
input = sys.stdin.readline

n = int(input())
nums = sorted(map(int, input().split()))
ans = 0

if n % 2:
    n -= 1
    ans = nums.pop()

for i in range(n // 2):
    ans = max(ans, nums[i] + nums[n - i - 1])

print(ans)